caikit/interfaces/nlp/data_model/text.py

```diff
@@ -0,0 +1,44 @@
+# Copyright The Caikit Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Data structures for text representations"""
+
+# Standard
+from typing import List
+
+# First Party
+from py_to_proto.dataclass_to_proto import Annotated, FieldNumber
+import alog
+
+# Local
+from ....core import DataObjectBase, dataobject
+from .package import NLP_PACKAGE
+
+log = alog.use_channel("DATAM")
+
+
+@dataobject(package=NLP_PACKAGE)
+class Token(DataObjectBase):
```

Comment:
@gkumbhat @alex-jw-brooks @tharapalanivel how does this definition of `token` compare to the general scientific literature? I've typically heard it used to describe "word" or "sub-word" parts, but never really heard it used to describe arbitrary spans. I can certainly see an argument for "a token is any atomic group of characters pertinent to the algorithm" so I'm ok using the word here if that's what we're going for.

Reply:
I think considering tokenization can also refer to "sentence tokenization", the broader definition of tokenization would sort of include breaking input text into spans based on some criteria. 

Example of sentence tokenizer: https://claritynlp.readthedocs.io/en/latest/developer_guide/algorithms/sentence_tokenization.html#:~:text=Sentence%20tokenization%20is%20the%20process,corpus%20of%20formal%20English%20text.